modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/validation/ParameterModelValidator.java

```diff
@@ -50,46 +61,87 @@
     @Override
     public void validate(ExtensionModel extensionModel) throws IllegalModelDefinitionException
     {
+        MetadataTypeVisitor visitor = new MetadataTypeVisitor()
+        {
+            private Set<Class<?>> visitedClasses = new HashSet<>();
+            private ClassTypeLoader typeLoader = ExtensionsTypeLoaderFactory.getDefault().createTypeLoader();
+
+            @Override
+            public void visitDictionary(DictionaryType dictionaryType)
+            {
+                dictionaryType.getKeyType().accept(this);
+                dictionaryType.getValueType().accept(this);
+            }
+
+            @Override
+            public void visitArrayType(ArrayType arrayType)
+            {
+                arrayType.getType().accept(this);
+            }
+
+            @Override
+            public void visitObject(ObjectType objectType)
+            {
+                Class<?> type = getType(objectType);
+                if (visitedClasses.add(type))
+                {
+                    for (Field f : type.getDeclaredFields())
```

Comment:
this is wrong. Here you're not considering inherited fields not any of the introspection rules likes aliases, ignores, etc. You're also ignoring Fields which are of a complex type. Use ObjectType#getFields() instead and make sure to also make this recursive on them too